t/python/t0010-job.py

```diff
@@ -784,6 +784,30 @@ def test_34_timeleft(self):
         except OSError:
             pass
 
+    def test_35_setattr_defaults(self):
+        """Test setattr setting defaults"""
+        jobspec = Jobspec.from_yaml_stream(self.basic_jobspec)
+        jobspec.setattr("cow", 1)
+        jobspec.setattr("system.cat", 2)
+        jobspec.setattr("user.dog", 3)
+        jobspec.setattr("attributes.system.chicken", 4)
+        jobspec.setattr("attributes.user.duck", 5)
+        jobspec.setattr("attributes.goat", 6)
```

Comment:
Add a test with leading `.` since that's explicitly handled in the method?

Reply:
recall that the leading `.` is only with `--setattr` not with this function.  We can definitely add a test ... i guess `.foo` turns into `attributes.system..foo`? (just thinking off the top of my head right now)

Reply:
Oh, yeah, I guess I confused myself with the various bits of code. Sorry.